neutron/live_test.go

```diff
@@ -0,0 +1,239 @@
+		c.Check(subnet.NetworkId, gc.Not(gc.Equals), "")
+		c.Check(subnet.Name, gc.Not(gc.Equals), "")
+		c.Assert(subnet.Cidr, gc.Matches, `\d{1,3}(\.+\d{1,3}){3}\/\d+`)
+	}
+	firstSubnet := subnets[0]
+	foundSubnet, err := s.neutron.GetSubnetV2(firstSubnet.Id)
+	c.Assert(err, gc.IsNil)
+	c.Check(foundSubnet.Id, gc.Equals, firstSubnet.Id)
+	c.Check(foundSubnet.NetworkId, gc.Equals, firstSubnet.NetworkId)
+	c.Check(foundSubnet.Name, gc.Equals, firstSubnet.Name)
+	c.Assert(foundSubnet.Cidr, gc.Matches, `\d{1,3}(\.+\d{1,3}){3}\/\d+`)
+}
+
+func (s *LiveTests) TestSecurityGroupsV2(c *gc.C) {
+	newSecGrp, err := s.neutron.CreateSecurityGroupV2("SecurityGroupTest", "Testing create security group")
+	c.Assert(err, gc.IsNil)
+	c.Assert(newSecGrp, gc.Not(gc.IsNil))
+	defer s.neutron.DeleteSecurityGroupV2(newSecGrp.Id)
+	secGrps, err := s.neutron.ListSecurityGroupsV2()
+	c.Assert(err, gc.IsNil)
+	c.Assert(secGrps, gc.Not(gc.HasLen), 0)
+	var found bool
+	for _, secGrp := range secGrps {
+		c.Check(secGrp.Id, gc.Not(gc.Equals), "")
+		c.Check(secGrp.Name, gc.Not(gc.Equals), "")
+		c.Check(secGrp.Description, gc.Not(gc.Equals), "")
+		c.Check(secGrp.TenantId, gc.Not(gc.Equals), "")
+		// Is this the SecurityGroup we just created?
+		if secGrp.Id == newSecGrp.Id {
+			found = true
+		}
+	}
+	if !found {
+		c.Errorf("expected to find added security group %s", newSecGrp)
+	}
+	// Change the created SecurityGroup's name
+	updatedSecGroup, err := s.neutron.UpdateSecurityGroupV2(newSecGrp.Id, "NameChanged", "")
+	// Verify the name change
+	foundSecGrps, err := s.neutron.SecurityGroupByNameV2(updatedSecGroup.Name)
+	c.Assert(err, gc.IsNil)
+	c.Assert(foundSecGrps, gc.Not(gc.HasLen), 0)
+	found = false
+	for _, secGrp := range foundSecGrps {
+		if secGrp.Id == updatedSecGroup.Id {
+			found = true
+			break
+		}
+	}
+	if !found {
+		c.Errorf("expected to find added security group %s, when requested by name", updatedSecGroup.Name)
+	}
+	_, err = s.neutron.SecurityGroupByNameV2(newSecGrp.Name)
+	c.Assert(err, gc.Not(gc.IsNil))
+	err = s.neutron.DeleteSecurityGroupV2(updatedSecGroup.Id)
+	c.Assert(err, gc.IsNil)
+}
+
+func (s *LiveTests) TestSecurityGroupsByNameV2(c *gc.C) {
+	// Create and find a SecurityGroup
+	newSecGrp, err := s.neutron.CreateSecurityGroupV2("SecurityGroupTest", "Testing find security group by name")
+	c.Assert(err, gc.IsNil)
+	defer s.neutron.DeleteSecurityGroupV2(newSecGrp.Id)
+	c.Assert(newSecGrp, gc.Not(gc.IsNil))
+	foundSecGrps, err := s.neutron.SecurityGroupByNameV2(newSecGrp.Name)
+	c.Assert(err, gc.IsNil)
+	c.Assert(foundSecGrps, gc.HasLen, 1)
+	if newSecGrp.Id != foundSecGrps[0].Id {
+		c.Errorf("expected to find added security group %s, when requested by name", newSecGrp.Name)
+	}
+	// Try to find a SecurityGroup that doesn't exist
+	errorSecGrps, err := s.neutron.SecurityGroupByNameV2("deadbeef")
+	c.Assert(err, gc.Not(gc.IsNil))
+	c.Assert(errorSecGrps, gc.HasLen, 0)
+	// Create and find a SecurityGroup with spaces in the name
```

Comment:
most of the tests could be broken up into multiple tests.  i'm going to leave this as is.